### PR TITLE
GNSS: Remove ambiguities

### PR DESCRIPTION
**Pull request by bjarki-andreasen**
Update the GNSS API and configs to remove ambiguities identified by #76364

This PR explicitly documents the expectation of leap seconds in the UTC based `struct gnss_time` structure, and adds the explicit configuration choice `GNSS_REFERENCE_FRAME_` for selecting the reference frame datum to use within the GNSS subsystem. All in tree drivers support and/or use WGS84 by default, (except for the nmea-generic driver where this is unknown, so WGS84 is presumed in this case).

GNSS modems which support setting the reference frame datum at runtime will need to configure this according to the `GNSS_REFERENCE_FRAME_` selection.

### CODE REVIEW

**Review by fabiobaltieri — APPROVED** [review]

File: include/zephyr/drivers/gnss.h [hunk 1]
```diff
@@ -169,7 +169,7 @@ struct gnss_time {
 	uint8_t hour;
 	/** Minute [0, 59] */
 	uint8_t minute;
-	/** Millisecond [0, 59999] */
+	/** Millisecond [0, 60999] */
 	uint16_t millisecond;
```
Comment (JordanYates) [inline include/zephyr/drivers/gnss.h:173]: What is the reasoning behind this field combining seconds and milliseconds? It seems out of place compared to everything else getting its own field.

This also drops a substantial amount of precision. GNSS modems report time accuracy in nanoseconds, and the RTC resolution of most systems will be ~30.5 microseconds from the 32768Hz LF XTAL.

Comment (JordanYates) [reply include/zephyr/drivers/gnss.h:173]: It just seems like an odd line to draw, especially when the struct is 7 bytes as is, so you're going to lose a byte to padding regardless.

Comment (bjarki-andreasen) [reply include/zephyr/drivers/gnss.h:173]: Do you want this member to be altered as part of this PR? Its not exactly related to the topic of ambiguity :)

Comment (JordanYates) [reply include/zephyr/drivers/gnss.h:173]: No, it was just convenient to ask here

**Review by JordanYates — APPROVED** [review]

**Comment by JordanYates** [comment]
This resolves the reference frame problem but not the altitude

**Review by tomi-font — APPROVED** [review]

**Pull request merged**